tap-agent/src/tap/sender_account.rs

```diff
@@ -189,34 +174,35 @@ impl Inner {
     }
 
     /// Recompute the sender's total unaggregated fees value and last receipt ID.
-    async fn recompute_unaggregated_fees_static(&self) -> Result<()> {
+    async fn recompute_unaggregated_fees(&self) -> Result<()> {
+        // Make sure to pause the handling of receipt notifications while we update the unaggregated
+        // fees.
+        let _guard = self.unaggregated_receipts_guard.lock().await;
+
         // Similar pattern to get_heaviest_allocation().
-        let allocations: Vec<_> = self.allocations.lock().await.values().cloned().collect();
+        let allocations: Vec<_> = self.allocations.lock().unwrap().values().cloned().collect();
 
-        let mut set = JoinSet::new();
+        let mut unaggregated_fees = self.unaggregated_fees.lock().unwrap();
+        *unaggregated_fees = UnaggregatedReceipts::default(); // Reset to 0.
         for allocation in allocations {
             let allocation: Arc<SenderAllocation> = match allocation {
                 AllocationState::Active(a) => a,
                 AllocationState::Ineligible(a) => a,
             };
-            set.spawn(async move { allocation.get_unaggregated_fees().await });
+
+            let uf = allocation.get_unaggregated_fees();
+            *unaggregated_fees = UnaggregatedReceipts {
+                value: self.fees_add(unaggregated_fees.value, uf.value),
+                last_id: max(unaggregated_fees.last_id, uf.last_id),
+            };
         }
 
         // Added benefit to this lock is that it pauses the handle_new_receipt_notification() calls
         // while we're recomputing the unaggregated fees value. Hopefully this is a very short
         // pause overall.
-        let mut unaggregated_fees = self.unaggregated_fees.lock().await;
 
         // Recompute the sender's total unaggregated fees value and last receipt ID, because new
         // receipts might have been added to the DB in the meantime.
```

Comment:
Could you update the location of these comments?

Reply:
Fix in 9c35d85203b854e292fd91c464cd4f1acbf15a2e